packages/core/src/builder.ts

```diff
@@ -614,4 +615,30 @@ export default class SchemaBuilder<Types extends SchemaTypes> {
       ? processedSchema
       : lexicographicSortSchema(processedSchema);
   }
+
+  withScalars<const ScalarRecord extends Record<string, GraphQLScalarType>>(scalars: ScalarRecord) {
+    const that = this as unknown as SchemaBuilder<
+      PothosSchemaTypes.ExtendDefaultTypes<
+        Types & {
+          Scalars: {
+            // Extract the Input and Output types from GraphQLScalarType's generics
+            [Property in keyof ScalarRecord]: ScalarRecord[Property] extends GraphQLScalarType<
+              infer TInternal,
+              infer TExternal
+            >
+              ? {
+                  Input: TInternal;
+                  Output: TExternal;
+                }
+              : never;
+          };
+        }
+      >
+    >;
+
+    for (const [name, scalar] of toPairs(scalars)) {
+      that.addScalarType(name, scalar, {});
```

Comment:
I know it's inconvenient and a bit awkward and inconvenient, but it's pretty important to be able to provide these options. 

This could be `Date: [DateResolver, options]` or `Date: { type/resolver/implementation: DateResolver, ...otherOptions }`

Personally I like the second option (although the resolver terminology from graphql-scalars is not my favorite.

I'd probably make it work so you can pass either just the scalar, or scalar + options.  Inference will be slightly more complex but shouldn't be too bad.

Reply:
I'm working on this, might take a bit of time

I am aiming to have the following be valid.
```ts
const builder = new SchemaBuilder({}).withScalars({
  Date: DateScalar,
  UUID: { scalar: UUIDScalar },
  PositiveInt: {
    scalar: PositiveIntScalar,
    options: {
      extensions: { codegenScalarType: "number" },
    },
  },
});
```

Does that look good to you?